lib/SILOptimizer/Mandatory/TFPartition.cpp

```diff
@@ -73,11 +73,15 @@ static InFlightDiagnostic diagnose(ASTContext &Context, SourceLoc loc,
 static bool isUserIgnoredByPartitioning(SILInstruction *inst) {
   auto optMode = inst->getFunction()->getEffectiveOptimizationMode();
   // `debug_value` instructions are used for value inspection during debugging.
-  // For optimized builds, we ignored these instructions so that no send/receive
-  // will be emitted.
-  if (isa<DebugValueInst>(inst) &&
-      optMode != OptimizationMode::NoOptimization) {
-    return true;
+  if (auto *DVI = dyn_cast<DebugValueInst>(inst)) {
+    // Ignore all debug_value instructions for an optimized build so that no
+    // send/receive will be triggered.
+    if (optMode != OptimizationMode::NoOptimization)
+      return true;
+    // Ignore opaque handle because they cannot be sent/received.
+    if (isOpaqueHandle(DVI->getOperand()->getType()))
```

Comment:
just write: `return isOpaqueHandle(DVI->getOperand()->getType();`?

Reply:
Sure, but I feel it's more readable to write `if`'s to be consistent with the previous few conditions. Changed now.